test-suite/src/test/scala/org/scalajs/testsuite/javalib/LongTest.scala

```diff
@@ -114,6 +114,36 @@ object LongTest extends JasmineTest {
       test("-24", -0x24L)
       test("30000", 0x30000L)
       test("-90000", -0x90000L)
+      test("bfc94973", 3217639795L)
+      test("bfc949733", 51482236723L)
+    }
+
+    it("should parse strings in bases 2 to 36") {
+      def test(radix: Int, s: String, v: Long): Unit = {
+        expect(JLong.parseLong(s, radix)).toEqual(v)
+        expect(JLong.valueOf(s, radix).longValue()).toEqual(v)
+      }
+      def testNumber(i: Int): Long = {
```

Comment:
Ok. As a side note: When writing the test, and the fact you have suggested the name "gen", I could not get the thought of scalacheck from my mind. In fact, it's the perfect tool for all these sorts of base classes. Just a thought